backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v2/order/sales/OrderService.java

```
@@ -0,0 +1,168 @@
+		}
+
+		final BankAccountId targetBankAccount = paymentService.determineInboundBankAccountId(orgId, currencyId, request.getTargetIBAN())
+				.orElseThrow(() -> new AdempiereException(String.format(
+						"Cannot find Bank Account for org-id: %s, currency: %s and iban: %s", orgId, currencyId, request.getTargetIBAN())));
+
+		final ExternalIdentifier bPartnerExternalIdentifier = ExternalIdentifier.of(request.getBpartnerIdentifier());
+		final BPartnerId bPartnerId = jsonRetrieverService.resolveBPartnerExternalIdentifier(bPartnerExternalIdentifier, orgId)
+				.orElseThrow(() -> new AdempiereException("No BPartner could be found for the given external BPartner identifier!")
+						.appendParametersToMessage()
+						.setParameter("externalBPartnerIdentifier", bPartnerExternalIdentifier.getRawValue())
+						.setParameter("orgId", orgId));
+
+		trxManager.runInThreadInheritedTrx(() -> {
+			final DefaultPaymentBuilder paymentBuilder = paymentService.newInboundReceiptBuilder()
+					.adOrgId(orgId)
+					.bpartnerId(bPartnerId)
+					.payAmt(request.getAmount())
+					.discountAmt(request.getDiscountAmt())
+					.writeoffAmt(request.getWriteOffAmt())
+					.currencyId(currencyId)
+					.orgBankAccountId(targetBankAccount)
+					.tenderType(TenderType.DirectDeposit)
+					.dateAcct(dateTrx)
+					.dateTrx(dateTrx)
+					.externalId(ExternalId.ofOrNull(request.getExternalPaymentId()));
+
+			final IdentifierString orderIdentifier = IdentifierString.of(request.getOrderIdentifier());
+
+			final Optional<OrderId> orderId = resolveOrderId(orderIdentifier, orgId);
+			orderId.ifPresent(paymentBuilder::orderId);
+
+			if (!orderId.isPresent() && IdentifierString.Type.EXTERNAL_ID.equals(orderIdentifier.getType()))
+			{
+				paymentBuilder.orderExternalId(orderIdentifier.asExternalId().getValue());
+			}
+			else if (!orderId.isPresent())
+			{
+				throw new AdempiereException("No order could be found for the given order identifier!")
```

please throw a MissingResourceException (subclass of AdempiereException) and include the org-id as detail (no need to localize that detail via AD_Message)